### PR TITLE
Improve JUnit Usage

### PR DESCRIPTION
**Pull request by marschall**
Improve the way JUnit is used. This commit includes the following
changes:

- update JUnit 4 version to latest
- replace custom Assert class with JUnit provided one
- switch from assertEquals to assertArrayEquals for arrays
- use fail() instead of assertFalse(true)
- used static import of asserts for better readability

### CODE REVIEW

**Comment by simmac** [comment]
Looks good, thank you!

**Comment by simmac** [comment]
I will notify Threema about the PRs I merged and reopen them for pulling them into the `official` branch

**Comment by marschall** [comment]
@simmac is that repo still maintained? I already opened PRs there
 * https://github.com/threema-ch/threema-msgapi-sdk-java/pull/13
 * https://github.com/threema-ch/threema-msgapi-sdk-java/pull/12

**Comment by simmac** [comment]
@marschall The threema-ch/threema-msgapi-sdk-java repo is not maintained anymore, unfortunately, the official version of the Threema Gateway Java API is available here though: https://gateway.threema.ch/de/developer/sdk-java

My `official` branch is just a mirror of the code which can be downloaded on the gateway.threema.ch page. I try to get all the changes we make in this "community enhanced version" of the msgapi into the official version by sending all patches to the Threema team. Once they release a new version on their site, I will merge all commits to the `official` branch, so that the 'official' branch is always an up-to-date mirror.

tl;dr: This repo is the only maintained one on GitHub. Official version is available at https://gateway.threema.ch/de/developer/sdk-java, the branch `simmac:official` is a mirror. 'simmac:master' is the "community enhanced" version, I try to get Threema to merge those enhancements into the official version.

**Comment by marschall** [comment]
That's for clearing that up for me.
